### ISSUE TITLE
Secure MQTT

### ISSUE THREAD

**Issue by jeremypoulter** [OpenEVSE/ESP32_WiFi_V3.x#31]
Add support for secure MQTT connections, depends on #29

**Comment by glynhudson** [comment]
Load certificates for:

- mqtt.emoncms.org
- io.adafruit.com
- Amazon MQ

**Comment by chris1howell** [comment]
and:

data.openevse.com

**Comment by glynhudson** [comment]
@chris1howell  Do you run a MQTT server at `data.openevse.com`? This issue was specially regarding MQTTS. Posting to data.openevse.com via HTTPS is already supported. I will confirm this with @jeremypoulter

**Comment by jeremypoulter** [comment]
Correct, posting to data.openevse.org via HTTPS is already supposed. MQTT is a slightly different usecase as you are more likely to be posting to an unknown service.

**Comment by chris1howell** [comment]
MQTT is running at data.openevse.com. We are setting ESP32 modules to MQTT
enabled by default using HAL.getLongID() as the default topic. We are using
a default user/password for now but would like to move to a unique client
cert per device.

**Comment by jeremypoulter** [comment]
I  think you are talking about client certificates right? This is not what
this covers, just connection via secure MQTT, the certs referenced are the
server root certificates.

**Comment by chris1howell** [comment]
Yes, the client certificate is for authorization only replacing a static
MQTT username and password.

SSL Encryption over MQTT is needed to data.openevse.com.

**Comment by jeremypoulter** [comment]
Ok let's add the client certificate support as another issue, and keep this as just connection to the server via MQTTS

**Comment by jeremypoulter** [comment]
@chris1howell can you send details of how to connect with a user/pass for testing?

**Comment by chris1howell** [comment]
Sure. I have dozens of 3.02 clients connected now. I will DM
username/password.

**Comment by jeremypoulter** [comment]
Have this working with Amazon MQ, but I am having issues with Adafruit and data.openevse.com, reporting as not supported :( connecting with OpenSSL to both of those uses TLS1.3 which is not supported but should fall back to TLS1.2.
